src/charm.py

```diff
@@ -182,6 +189,45 @@ def _on_config_changed(self, _: ops.HookEvent, charm_state: CharmState) -> None:
         self.change_config(charm_state)
         self._set_workload_version()
 
+    @inject_charm_state
+    def _on_relation_joined(self, _: ops.HookEvent, charm_state: CharmState) -> None:
+        """Handle Synapse peer relation joined event.
+
+        Args:
+            charm_state: The charm state.
+        """
+        if charm_state.redis_config is None:
+            logger.debug("Scaling is not allowed without Redis integration.")
```

Comment:
Won't this happen also if there is only one unit in the application?

Reply:
I hope not. I'm considering this:
“For a peer relation, <peer relation name>-relation-joined will only be emitted if the scale is larger than 1. In other words, applications with a scale of 1 do not see peer relation joined/departed events. If you are using peer data as a means for persistent storage, then use peer relation-created instead.”
https://juju.is/docs/sdk/relation-name-relation-joined-event

Reply:
TIL